internal/debaml/bamlunicode/testdata/reference/src/main.rs

```
@@ -0,0 +1,231 @@
+fn to_hex(bytes: &[u8]) -> String {
+    let mut s = String::with_capacity(bytes.len() * 2);
+    for b in bytes {
+        s.push(char::from_digit((b >> 4) as u32, 16).unwrap());
+        s.push(char::from_digit((b & 0xf) as u32, 16).unwrap());
+    }
+    s
+}
+
+fn from_hex(s: &str) -> Vec<u8> {
+    let b = s.trim().as_bytes();
+    assert!(b.len() % 2 == 0, "hex input must have even length");
```

Clippy (1.96.0)

[warning] manual implementation of `.is_multiple_of()`

(warning)